From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct sdio_func
Patch-mainline: never, kabi
References: kabi

In 4.4.91, commit 1b760fdad9f0626e7be2c1082c94dca21f72470b (mmc: sdio:
fix alignment issue in struct sdio_func), upstream commit
5ef1ecf060f28ecef313b5723f1fd39bf5a35f56 changed the tmpbuf member from
u8[4] to u8 *.  This indeed changed the layout of the structure and the
kABI checker now complains.

Given this structure is allocated only in the core by sdio_alloc_func
and tmpbuf is used only by the core, move the member with new type to
the end and make the change invisible to the kABI checker.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/mmc/sdio_func.h |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

--- a/include/linux/mmc/sdio_func.h
+++ b/include/linux/mmc/sdio_func.h
@@ -53,12 +53,19 @@ struct sdio_func {
 	unsigned int		state;		/* function state */
 #define SDIO_STATE_PRESENT	(1<<0)		/* present in sysfs */
 
-	u8			*tmpbuf;	/* DMA:able scratch buffer */
+#ifdef __GENKSYMS__
+	u8			tmpbuf[4];	/* DMA:able scratch buffer */
+#else
+	u8			kabi_ph[4];	/* kabi placeholder */
+#endif
 
 	unsigned		num_info;	/* number of info strings */
 	const char		**info;		/* info strings */
 
 	struct sdio_func_tuple *tuples;
+#ifndef __GENKSYMS__
+	u8			*tmpbuf;	/* DMA:able scratch buffer */
+#endif
 };
 
 #define sdio_func_present(f)	((f)->state & SDIO_STATE_PRESENT)
